sillsdev/web-languageforge
# Audio settings

Fixes #1508 

## Description

We are adding two new project settings: options for codecs for audio recording, and options for whether/when to convert uploaded audio. 

Audio codec:
WEBM/OPUS - default
WAV/PCM

Right now, all recordings  are given to us in webm containers from MediaRecorder, even though pcm codec is specified for recording wav files. So, we are converting to wav during upload.

When to convert audio:
Always
Only if necessary for Send/Receive ( = files larger than 1MB (current limit), or file type is other than .webm, .mp3, or .wav)
Never - default

### Type of Change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update - potentially
- UI change

## Screenshots
![Screenshot (8)](https://user-images.githubusercontent.com/56163492/206118500-6909958a-2c7e-4be7-8db3-e4a3e0cc1162.png)

![Screenshot (9)](https://user-images.githubusercontent.com/56163492/206118625-59743eeb-3712-445c-9578-76b3f06b9631.png)

## Checklist


## How to test

[Audio Test Upload Files.zip](https://github.com/sillsdev/web-languageforge/files/10129280/Audio.Test.Upload.Files.zip)
Here is a folder of test audio files. I refer them in the test cases below...

- First, set the "Convert uploaded audio to WEBM..." setting to "Never". Upload an .ogg/.flac/.m4a  file; observe that it is NOT converted. Upload the long WAV file; observe that it is NOT converted. 
- Then, set the "Convert uploaded audio to WEBM..." setting to "Only if necessary for Send/Receive".  Upload a short .wav file. Observe that it is NOT converted. Upload a short .ogg file or .flac file or .m4a file. Observe that it is converted. Upload any long file. Observe that it is converted. 
- Then, set the 'Convert uploaded audio to WEBM..." setting to "Always". Upload a short .wav file and/or .mp3 file. Observe that it is converted. Upload any other files; they should all be converted to .webm.
- Set the "Audio recording codec" setting to WEBM/OPUS. Record something and save it. Click the three dots beside the playback and view the file; it should have a .webm extension.
- Set the "Audio recording codec" setting to PCM/WAV. Record something and save it. Click the three dots beside the playback and view the file; it should have a .wav extension.

## qa.languageforge.org testing

Testers should add his/her findings to end of the PR in a comment and include screenshots, files, etc that are beneficial.
